modin/engines/base/frame/data.py

```diff
@@ -141,11 +141,9 @@ def _compute_dtypes(self):
         def dtype_builder(df):
             return df.apply(lambda col: find_common_type(col.values), axis=0)
 
-        map_func = self._build_mapreduce_func(0, lambda df: df.dtypes)
-        reduce_func = self._build_mapreduce_func(0, dtype_builder)
         # For now we will use a pandas Series for the dtypes.
         if len(self.columns) > 0:
-            dtypes = self._map_reduce(0, map_func, reduce_func).to_pandas().iloc[0]
+            dtypes = self._map_reduce(0, lambda df: df.dtypes, dtype_builder)
```

Comment:
This is incorrect, it will always return `object` for every column because the dtype of a `Series` of dtypes is `object`. Is this the cause of the performance?

Reply:
I have an example of dtypes output:

```python

(Pdb) test.df.dtypes
col0       float64
col1       float64
col2       float64
col3       float64
col4       float64
            ...
col4995    float64
col4996    float64
col4997    float64
col4998    float64
col4999    float64
Length: 5000, dtype: object
(Pdb) type(test.df.dtypes)
<class 'pandas.core.series.Series'>

```

Reply:
Oh, I forgot to remove some of the code

Reply:
Yes, sorry. I missed the third argument `dtype_builder`